### PR TITLE
CR-1142391 - Programming xclbin immediately after xbutil reset fails on V70/VCK5000

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
Signed-off-by: rbramand <rbramand@xilinx.com>

#### Problem solved by the commit
xbutil reset triggers reboot of apu and boot services are relaunched. 
apu-boot service writes to a register indicating that it is ready. Programming xclbin from host just after reset fails because soft kernel daemon which is started by another boot service is not finished yet but apu-boot service indicated that system is ready. This pr solves this synchronization issue.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
It was discovered in local regression testing by filer

#### How problem was solved, alternative solutions (if any) and why they were rejected
Added skd service as dependency to apu-boot service which will be started only after skd service is finished.

#### Risks (if any) associated the changes in the commit
Low

#### What has been tested and how, request additional testing if necessary
Tested by running regression with the newly generated apu package on V70 device.

#### Documentation impact (if any)
NA

### CODE REVIEW

File: src/runtime_src/tools/scripts/apu_recipes/skd.service [hunk 1]
```diff
@@ -1,10 +1,8 @@
 [Unit]
 Description=Soft Kernel Daemon
-After=apu-boot.service
 
 [Service]
 Type=forking
-User=softkernel
```
Comment (chvamshi-xilinx) [inline src/runtime_src/tools/scripts/apu_recipes/skd.service:7]: why did we remove the user?
@jeffli-xilinx , please review this change

Comment (jeffli-xilinx) [reply src/runtime_src/tools/scripts/apu_recipes/skd.service:7]: @rbramand-xilinx 
Please keep User=softkernel

You can modify apu-boot and move all the lines that come after rpm-channel/ready to before the if statement.

Original:
if [ -e /sys/bus/platform/devices/rpu-channel/ready ]; then
	echo 1 > /sys/bus/platform/devices/rpu-channel/ready
	# Add softkernel user for PS kernel daemon to run as
	useradd softkernel
	# Work-around for AIE as it defaults root access only
	chmod 666 /dev/aie0
	# Create default xrt.ini
	echo "# xrt.ini generated by apu-boot script" > /usr/bin/xrt.ini
	echo "[Runtime]" >> /usr/bin/xrt.ini
	echo "verbosity=5" >> /usr/bin/xrt.ini
	echo "runtime_log=syslog" >> /usr/bin/xrt.ini
fi

Update:
# Add softkernel user for PS kernel daemon to run as
useradd softkernel
# Work-around for AIE as it defaults root access only
chmod 666 /dev/aie0
# Create default xrt.ini
echo "# xrt.ini generated by apu-boot script" > /usr/bin/xrt.ini
echo "[Runtime]" >> /usr/bin/xrt.ini
echo "verbosity=5" >> /usr/bin/xrt.ini
echo "runtime_log=syslog" >> /usr/bin/xrt.ini
if [ -e /sys/bus/platform/devices/rpu-channel/ready ]; then
	echo 1 > /sys/bus/platform/devices/rpu-channel/ready
fi

**Review by jeffli-xilinx — CHANGES_REQUESTED** [review]
@rbramand-xilinx 
Please keep User=softkernel

You can modify apu-boot and move all the lines that come after rpm-channel/ready to before the if statement.

Original:
if [ -e /sys/bus/platform/devices/rpu-channel/ready ]; then
	echo 1 > /sys/bus/platform/devices/rpu-channel/ready
	# Add softkernel user for PS kernel daemon to run as
	useradd softkernel
	# Work-around for AIE as it defaults root access only
	chmod 666 /dev/aie0
	# Create default xrt.ini
	echo "# xrt.ini generated by apu-boot script" > /usr/bin/xrt.ini
	echo "[Runtime]" >> /usr/bin/xrt.ini
	echo "verbosity=5" >> /usr/bin/xrt.ini
	echo "runtime_log=syslog" >> /usr/bin/xrt.ini
fi

Update:
\# Add softkernel user for PS kernel daemon to run as
useradd softkernel
\# Work-around for AIE as it defaults root access only
chmod 666 /dev/aie0
\# Create default xrt.ini
echo "# xrt.ini generated by apu-boot script" > /usr/bin/xrt.ini
echo "[Runtime]" >> /usr/bin/xrt.ini
echo "verbosity=5" >> /usr/bin/xrt.ini
echo "runtime_log=syslog" >> /usr/bin/xrt.ini
if [ -e /sys/bus/platform/devices/rpu-channel/ready ]; then
	echo 1 > /sys/bus/platform/devices/rpu-channel/ready
fi

**Comment by chvamshi-xilinx** [comment]
retest this please

File: src/runtime_src/tools/scripts/apu_recipes/init-apu.bb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+SUMMARY = "PS Kernel Daemon"
```
Comment (jeffli-xilinx) [inline src/runtime_src/tools/scripts/apu_recipes/init-apu.bb:1]: Please update the summary here.
Thanks!

File: src/runtime_src/tools/scripts/apu_recipes/init-apu.service [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=APU init Daemon
+
+[Service]
+ExecStart=/usr/bin/init-apu
+StandardOutput=journal+console
+MemoryLimit=1G
```
Comment (jeffli-xilinx) [inline src/runtime_src/tools/scripts/apu_recipes/init-apu.service:7]: Is MemoryLimit needed here?
If so, it should probably be less than 1G.

**Review by jeffli-xilinx — CHANGES_REQUESTED** [review]
Just some minor updates requested.
Thanks!

File: src/runtime_src/tools/scripts/apu_recipes/init-apu.service (hunk 3 above)
Comment (rbramand-xilinx) [reply src/runtime_src/tools/scripts/apu_recipes/init-apu.service]: Thanks @jeffli-xilinx for pointing it, memory limit is not needed. Made changes please approve

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by jeffli-xilinx — APPROVED** [review]

**Comment by chvamshi-xilinx** [comment]
Please mark the CR fixed and backport to 2022.2

**Pull request merged**